src/graph.py

```diff
@@ -635,6 +644,54 @@ def grid_search(self, prize_file, Gs, Bs, Ws):
         return forest, augmented_forest, params_by_nodes
 
 
+    def _eval_randomizations(self, params):
+        """
+        Convenience method which sets parameters and performs PCSF
+        """
+
+        self._reset_hyperparameters(params)
+        paramstring = "G_{}_B_{}_W_{}".format(*[int(x) if int(x) == x else x for x in [params['g'], params['b'], params['w']]])
+        logger.info("Randomizations for " + paramstring)
+
+        forest, augmented_forest = self.randomizations(params["noisy_edges_repetitions"], params["random_terminals_repetitions"])
+        
+        return paramstring, forest, augmented_forest
+
+
+    def _grid_randomization(self, prize_file, params):
+        """
+        Internal function which executes pcsf at every point in a parameter grid.
+        Subroutine of `grid_search`.
+
+        Arguments:
+            prize_file (str): filepath
+            params (dict): params with which to run the program
+
+        Returns:
+            results (dict)
+        """
+
+        # get number of cpus available to job
+        try:
+            n_cpus = int(os.environ["SLURM_JOB_CPUS_PER_NODE"])
+        except KeyError:
+            n_cpus = multiprocessing.cpu_count()
+
+        pool = multiprocessing.Pool(n_cpus)
+
+        self.prepare_prizes(prize_file)
+
+        model_params = [{'w': w, 'b': b, 'g':g} for (w, b, g) in product(params['w'], params['b'], params['g'])]
+        other_params = {key: params[key] for key in params if key not in 'wbg'} # `not in 'wbg'` is a shortened way of writing `key != 'w' & key != 'b' & key != 'g'`
+        all_params = [{**model_param, **other_params} for model_param in model_params]
```

Comment:
having `model_params` and `other_params` seem parallel in lines 684-685 is misleading if in fact `model_params` is a list of permutations of model params whereas `other_params` is just a single parameter set.

Reply:
fe6e13ad29018b368cf28bd2fa455ccf8179a90e